### PR TITLE
Add additional OIDC Issuer query parameters

### PR DESCRIPTION
**Pull request by ashleywang1**
Allow additional Issuer query parameters to be appended to the authorization request url. The base url will still be the IssuerUrl, with all the same information appended to the end such as the client_id, scope, callback path, etc. This change allows additional parameters to be appended to the end of that string in the form &key=value.
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/2030

### CODE REVIEW

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 1]
```diff
@@ -204,6 +204,9 @@ message OAuth {
   // ".well-known/openid-configuration"
   string issuer_url = 3;
 
+  // The query parameters to apply to the call to the identity provider.
+  map<string, string> issuer_query_params = 7;
```
[the remark this answers is not in the document]
Comment (kdorosh) [reply projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto]: Initial thoughts, that's a long name. Still thinking of something better, but I agree that the name should align with OAuth terminology

[the remark this answers is not in the document]
Comment (ashleywang1) [reply projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto]: auth_endpoint_params?
oauth_endpoint_params?
just endpoint_params?

https://tools.ietf.org/html/rfc6749#page-50

Googling "oauth endpoint parameters" returns a list of the appropriate things like "client_id" and "scope", so I think parameters is a good word here.

I haven't seen "query" used anywhere in the OAuth docs.

Additionally, our Discovery object (internal code) is what deals with resolving the Endpoint URL. "endpoint" might make sense to someone familiar with the docs, but it's not intuitive to someone new.

[the remark this answers is not in the document]
Comment (kdorosh) [reply projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto]: i think it's worth noting that we already have `issuer_url` in the API; thus the name `issuer_query_params` may actually be the least misleading name.

issuer is a term used here, for example, see https://sitbfo19-secommunities.cs17.force.com/identity/.well-known/openid-configuration and search for "issuer"

[the remark this answers is not in the document]
Comment (marcogschmidt) [reply projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto]: I just want to make it explicit that these parameters are specific to the request that gloo sends to the authorization endpoint. So maybe your first suggestion (`auth_endpoint_params`) is the best one @ashleywang1 .

**Pull request merged**